inject/src/main/java/io/micronaut/inject/ast/ElementAnnotationMetadata.java

```diff
@@ -13,7 +13,23 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+package io.micronaut.inject.ast;
+
+import io.micronaut.core.annotation.AnnotationMetadata;
+
 /**
- * Package to organize classes responsible for validating executable methods at compile time.
+ * Element's annotation metadata.
+ *
+ * @author Denis Stepanov
+ * @since 4.0.0
  */
-package io.micronaut.validation.executable;
+public interface ElementAnnotationMetadata extends MutableAnnotatedElement<AnnotationMetadata> {
+
+    /**
+     * Gets the annotation metadata for the element.
+     *
+     * @return the annotation metadata
+     */
+    AnnotationMetadata get();
```

Comment:
Rename to `getAnnotationMetadata` and extend from `AnnotationMetadataProvider` would be better. Or could we not create an interface like `MutableAnnotationMetadataProvider` which would be more consistent with the rest of the API?

Reply:
Not sure I understand